### PR TITLE
Filter checks by probe

### PR DESCRIPTION
**Pull request by jewbetcha**
- adds a select to CheckList to filter by probes, with AND operator
- namespace CheckList with test file and some helpers

### CODE REVIEW

File: __mocks__/@grafana/ui.tsx [hunk 1]
```diff
@@ -40,11 +40,20 @@ const MultiSelect = forwardRef((props, ref) => <Select {...props} ref={ref} mult
 
 MultiSelect.displayName = 'MultiSelect';
 
+// Using this specifically for Probe filter at the moment
```
Comment (rdubrock) [inline __mocks__/@grafana/ui.tsx:43]: So now that I think about it, if you're mocking the datasource in your tests (which I think you are), then we _could_ theoretically just call the `loadOptions` prop and pass that as options to the mocked select. Probably further down the rabbit hole than we need to go right now, just thinking out loud

File: src/components/CheckList/CheckList.tsx [hunk 2]
```diff
@@ -540,6 +514,22 @@ export const CheckList = ({ instance, onAddNewClick, checks, onCheckUpdate }: Pr
             setCurrentPage(1);
           }}
           labelFilters={labelFilters}
+          className={styles.marginRightSmall}
+        />
+        <AsyncMultiSelect
+          data-testid="probe-filter"
+          prefix="Probes"
+          onChange={(v) => {
```
Comment (rdubrock) [inline src/components/CheckList/CheckList.tsx:522]: [nit] could this just be `onChange={setSelectedProbes}`?

File: src/page/ChecksPage.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import React, { PureComponent } from 'react';
 import { Check } from 'types';
 import { getLocationSrv } from '@grafana/runtime';
 import { CheckEditor } from 'components/CheckEditor';
-import { CheckList } from 'components/CheckList';
+import { CheckList } from 'components/CheckList/CheckList';
```
Comment (rdubrock) [inline src/page/ChecksPage.tsx:8]: You could put an `index.ts` that manages the exports from that directory if you like so the imports don't have to change

**Review by rdubrock — APPROVED** [review]
Looks great 🦏

File: src/components/CheckList/CheckList.tsx (hunk 2 above)
Comment (jewbetcha) [reply src/components/CheckList/CheckList.tsx:522]: Will try!

**Pull request merged**